src/linear_solver_logs/solve_log_full_ma.jl

```diff
@@ -0,0 +1,248 @@
+                          0,
+                          sigma2,
+                          omega,
+                          eta, 
+                          true_res
+                         )
+
+#Function to update the moving average
+function log_update!(
+    log::LSLogFullMA,
+    sampler::LinSysSampler,
+    x::AbstractVector,
+    samp::Tuple,
+    iter::Int64,
+    A::AbstractArray,
+    b::AbstractVector
+)
+    log.max_dimension = maximum(size(A))
+    ma_info = log.ma_info
+    log.iterations = iter
+    log.sampler = typeof(sampler)  
+    if iter != 0
+        #Check if we want exact residuals computed
+        if !log.true_res
+            # Compute the current residual to second power to align with theory
+            res::Float64 = log.resid_norm(samp[3])^2 
+        elseif typeof(sampler) <: LinSysVecColSampler 
+            res = log.resid_norm(A'* (A * x - b))^2
+        else
+            res = log.resid_norm(A * x - b)^2 
+        end
+        # Check if MA is in lambda1 or lambda2 regime
+        if ma_info.flag
+            Update_MAEstimators!(log, ma_info, res, ma_info.lambda2, iter)
+        else
+            #Check if we can switch between lambda1 and lambda2 regime
+            if res < ma_info.res_window[ma_info.idx]
+                Update_MAEstimators!(log, ma_info, res, ma_info.lambda1, iter)
+            else
+                Update_MAEstimators!(log, ma_info, res, ma_info.lambda1, iter)
+                ma_info.flag = true 
+            end
+
+        end
+
+    end
+
+end
+
+# Update the moving average estimator requires the log variable, ma_info,
+# observed residual, and a lambda_base which corresponds to which lambda regime we are in. 
+function Update_MAEstimators!(log, ma_info, res, lambda_base, iter)
+    accum = 0
+    accum2 = 0
+    ma_info.idx = ma_info.idx < ma_info.lambda2 ? ma_info.idx + 1 : 1
+    ma_info.res_window[ma_info.idx] = res
+    #Check if entire storage buffer can be used
+    if ma_info.lambda == lambda_base 
+        # Compute the moving average
+        @simd for i in 1:ma_info.lambda2
+            accum += ma_info.res_window[i]
+            accum2 += ma_info.res_window[i]^2
+        end
+       
+        if mod(iter, log.collection_rate) == 0 || iter == 0
+            push!(log.width_hist, ma_info.lambda)
+            push!(log.resid_hist, accum / ma_info.lambda) 
+            push!(log.iota_hist, accum2 / ma_info.lambda) 
+        end
+
+    else
+        # Get the difference between the start and current lambda
+        diff = ma_info.idx - ma_info.lambda
+        
+        # Determine start point for first loop
+        startp1 = diff < 0 ? 1 : (diff + 1)
+        
+        # Determine start and endpoints for second loop
+        startp2 = diff > 0 ? 2 : lambda_base + diff + 1  
+        endp2 = diff > 0 ? 1 : lambda_base
+
+        # Compute the moving average two loop setup required when lambda < lambda2
+        @simd for i in startp1:ma_info.idx
```

Comment:
Do we want to use these simd commands right now?

Reply:
I removed the simd